src/main/java/lv/ctco/javaschool/app/boundary/TripApi.java

```diff
@@ -74,13 +57,40 @@ public ListTripDto getActiveTrips() {
         return listTripDto;
     }
 
+    private TripDto convertToTripDto(Trip trip) {
+        User driver = trip.getDriver();
+        TripDto dto = new TripDto();
+        dto.setId(trip.getId());
+        dto.setDriverInfo(driver.getSurname() + " " + driver.getName());
+        dto.setDriverPhone(driver.getPhoneNumber());
+        dto.setEvent(trip.isEvent());
+        dto.setFrom(trip.getDeparture());
+        dto.setTo(trip.getDestination());
+        dto.setPlaces(trip.getPlaces());
+        dto.setTime(trip.getDepartureTime());
+        dto.setTripStatus(trip.getTripStatus());
+        return dto;
+    }
+
     @POST
     @Path("/{id}")
     @RolesAllowed({"ADMIN", "USER"})
-    public void setTripPlaces(JsonObject field, @PathParam("id") String tripId) {
-        for (Map.Entry<String, JsonValue> pair : field.entrySet()) {
-            Integer places = ((JsonNumber) pair.getValue()).intValue();
-            tripStore.setTripPlaces(places, tripId);
+    public Response setTripPlacesAndUser(JoinTripDto joinTripDto, @PathParam("id") Long tripId) {
+        User user = userStore.getCurrentUser();
+        Optional<Trip> tripOptional = tripStore.findTripById(tripId);
+        if (tripOptional.isPresent()) {
+            Trip trip = tripOptional.get();
+            if (trip.getPassengers().contains(user)) {
+                return Response.status(Response.Status.METHOD_NOT_ALLOWED).build();
```

Comment:
1. Unfortunately method not allowed too specific for this case. Usually it isn't used for business logic management. Use **400 Bad Request** or **409 Conflict** instead.
1. Please unify validations: use exceptions and exception mappers **OR** low-level management via returning Response object.

Note: you could use single exception for handling all validations. Just rename `TripNotFoundException` to `ValidationException` and add field with string message what gone wrong. In exception mapper return _Bad Request_ with JSON in the body with message.